### PR TITLE
Check for NA in DataArray inner constructor

### PR DESCRIPTION
**Pull request by bkamins**
Corrected PR JuliaStats/DataArrays.jl#189.

### CODE REVIEW

File: src/dataarray.jl [hunk 1]
```diff
@@ -22,12 +22,22 @@ type DataArray{T, N} <: AbstractDataArray{T, N}
...
 
-    # Ensure data values and missingness metadata match
     function DataArray(d::Array{T, N}, m::BitArray{N})
+        # Ensure data values and missingness metadata match
         if size(d) != size(m)
             msg = "Data and missingness arrays must be the same size"
             throw(ArgumentError(msg))
         end
+        # additionally check if d does not contain NA entries
+        if eltype(d) == Any
```
Comment (nalimilan) [inline src/dataarray.jl]: Sorry for not spotting this earlier, but better use `===` here. This should allow type inference to remove the branch in most cases.

File: src/dataarray.jl [hunk 2]
```diff
@@ -22,12 +22,22 @@ type DataArray{T, N} <: AbstractDataArray{T, N}
...
     na::BitArray{N}
 
-    # Ensure data values and missingness metadata match
     function DataArray(d::Array{T, N}, m::BitArray{N})
+        # Ensure data values and missingness metadata match
         if size(d) != size(m)
             msg = "Data and missingness arrays must be the same size"
             throw(ArgumentError(msg))
         end
+        # additionally check if d does not contain NA entries
```
Comment (nalimilan) [inline src/dataarray.jl]: "check that"

**Comment by bkamins** [comment]
@nalimilan thx for comments. I do not understand why on v0.5 tests started to fail in `reduce.jl`.

**Comment by nalimilan** [comment]
That failure is fixed with https://github.com/JuliaStats/DataArrays.jl/pull/192 AFAIK.

**Pull request merged**